src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas1/Cas1SpaceBookingService.kt

```diff
@@ -0,0 +1,75 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.service.cas1
+
+import kotlinx.datetime.toKotlinDatePeriod
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.ApprovedPremisesEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.Cas1SpaceBookingEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.Cas1SpaceBookingRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.cas1.Cas1SpaceSearchRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.validated
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.results.ValidatableActionResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.PlacementRequestService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.PremisesService
+import java.time.LocalDate
+import java.time.OffsetDateTime
+import java.util.UUID
+
+@Service
+class Cas1SpaceBookingService(
+  private val premisesService: PremisesService,
+  private val placementRequestService: PlacementRequestService,
+  private val cas1SpaceBookingRepository: Cas1SpaceBookingRepository,
+  private val cas1SpaceSearchRepository: Cas1SpaceSearchRepository,
+) {
+  fun createNewBooking(
+    premisesId: UUID,
+    placementRequestId: UUID,
+    arrivalDate: LocalDate,
+    departureDate: LocalDate,
+    createdBy: UserEntity,
+  ): ValidatableActionResult<Cas1SpaceBookingEntity> = validated {
+    val premises = premisesService.getPremises(premisesId) as? ApprovedPremisesEntity
+    if (premises == null) {
+      "$.premisesId" hasValidationError "doesNotExist"
+    }
+
+    val placementRequest = placementRequestService.getPlacementRequestOrNull(placementRequestId)
+    if (placementRequest == null) {
+      "$.placementRequestId" hasValidationError "doesNotExist"
+    }
+
+    if (arrivalDate >= departureDate) {
+      "$.departureDate" hasValidationError "beforeBookingArrivalDate"
```

Comment:
i'm not sure if we're constrained by existing validation error 'codes', but i think 'shouldBeAfterArrivalDate' would be a better description for this given we're doing a >= check?

Reply:
I think there's a list in the UI that these are mapped to, but I've changed it as it's more descriptive.